test/odbc/psqlodbc/test_pg_bpchar.cpp

```diff
@@ -0,0 +1,724 @@
+  verifyValuesInObject(ServerType::PSQL, VIEW_NAME, COL1_NAME, INSERTED_VALUES, EXPECTED_VALUES);
+
+  dropObject(ServerType::PSQL, "VIEW", VIEW_NAME);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<string> INSERTED_VALUES_4000 = {
+    "NULL", // NULL values
+    STRING_4000,
+    "" // blank value
+  };
+
+  const vector<string> EXPECTED_VALUES_4000 = {
+    "NULL", // NULL values
+    STRING_4000,
+    "" + std::string(4000, ' ')// blank value
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_4000);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, INSERTED_VALUES_4000, EXPECTED_VALUES_4000);
+
+  createView(ServerType::PSQL, VIEW_NAME, VIEW_QUERY);
+  verifyValuesInObject(ServerType::PSQL, VIEW_NAME, COL1_NAME, INSERTED_VALUES_4000, EXPECTED_VALUES_4000);
+
+  dropObject(ServerType::PSQL, "VIEW", VIEW_NAME);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<string> INSERTED_VALUES_20 = {
+    "NULL", // NULL values
+    STRING_20,
+    "" // blank value
+  };
+
+  const vector<string> EXPECTED_VALUES_20 = {
+    "NULL", // NULL values
+    STRING_20,
+    "" + std::string(20, ' ')// blank value
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_20);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, INSERTED_VALUES_20, EXPECTED_VALUES_20);
+
+  createView(ServerType::PSQL, VIEW_NAME, VIEW_QUERY);
+  verifyValuesInObject(ServerType::PSQL, VIEW_NAME, COL1_NAME, INSERTED_VALUES_20, EXPECTED_VALUES_20);
+
+  dropObject(ServerType::PSQL, "VIEW", VIEW_NAME);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Bpchar, Table_Single_Primary_Keys) {
+
+  const vector<pair<string, string>> TABLE_COLUMNS_1 = {
+    {COL1_NAME, "INT"},
+    {COL2_NAME, DATATYPE + "(1)"}
+  };
+
+  const string PKTABLE_NAME = TABLE_NAME.substr(TABLE_NAME.find('.') + 1, TABLE_NAME.length());
+  const string SCHEMA_NAME = TABLE_NAME.substr(0, TABLE_NAME.find('.'));
+
+  const vector<string> PK_COLUMNS = {
+    COL2_NAME
+  };
+
+  string tableConstraints = createTableConstraint("PRIMARY KEY ", PK_COLUMNS);
+
+  const vector<string> INSERTED_VALUES = {
+    STRING_1
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_1, tableConstraints);
+  testPrimaryKeys(ServerType::PSQL, SCHEMA_NAME, PKTABLE_NAME, PK_COLUMNS);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
+  testInsertionFailure(ServerType::PSQL, TABLE_NAME, COL1_NAME, INSERTED_VALUES, false, 0, false);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<pair<string, string>> TABLE_COLUMNS_4000 = {
+    {COL1_NAME, "INT"},
+    {COL2_NAME, DATATYPE + "(4000)"}
+  };
+
+  const vector<string> PK_COLUMNS_4000 = {
+    COL2_NAME
+  };
+
+  tableConstraints = createTableConstraint("PRIMARY KEY ", PK_COLUMNS_4000);
+
+  const vector<string> INSERTED_VALUES_4000 = {
```

Comment:
Rather than calling this `INSERTED_VALUES_4000`, changing it to `INSERTED_VALUES_2704` would be clearer.
Saying it 4000 is misleading